pkg/sql/sem/asof/as_of.go

```diff
@@ -204,6 +213,26 @@ func Eval(
 	return ret, nil
 }
 
+// asOfTypeResolver is a type resolver that always returns an error. It is used
+// to block type resolution while evaluating the AS OF SYSTEM TIME expression.
+type asOfTypeResolver struct {
+	err error
+}
+
+var _ tree.TypeReferenceResolver = (*asOfTypeResolver)(nil)
+
+// ResolveType implements the tree.TypeReferenceResolver interface.
+func (r *asOfTypeResolver) ResolveType(
+	ctx context.Context, name *tree.UnresolvedObjectName,
+) (*types.T, error) {
+	return nil, r.err
```

Comment:
nit: r.err will always be `nil`, correct? Can we `return nil, nil` here instead and remove the `err` field from the struct? And then `asOfTypeResolver` can implement the interface, instead of `*asOfTypeResolver`.

Reply:
Oops, disregard, I missed `err` being set above.